### PR TITLE
Adds passive voice assessment for French

### LINKED ISSUE #1286: Passive voice for French
We should add a passive voice assessment for French.

### PR DESCRIPTION
**Pull request by manuelaugustin**
## Summary

This PR can be summarized in the following changelog entry:

Adds a passive voice assessment for French.

## Relevant technical choices:

* Moves functionality from the English passive voice assessement to a shared `passivevoice` directory.

## Test instructions

This PR can be tested by following these steps:

- Check out the branch.
- In `src/assessments/readability/passiveVoiceAssessment.js` add `"fr"` to the available languages.
- Use the browserified example. Don't forget to run a `grunt build:js`.
-  Set the locale to `fr_FR`.
- Make sure the following sentences are marked as passive:
  - `J'étais mangé.` (passive sentence with regular participle)
  - `J’ai été mangé.` (passive sentence with the auxiliary `été`)
  - `Il était vendu.` (passive sentence with irregular participle)
  - `Elle était vendue.` (passive sentence with an irregular passive plus suffix)
  - `Il était mû.` (passive sentence with an irregular participle with irregular conjugation pattern)
  - `Une réponse a été promise à maintes reprises.` (passive sentence with an irregular participle ending in -s)
  - `Le jury de thèse de Mme Randerson était-il informé de cela ?` (passive sentence with an inverted auxiliary in a question)

- Make sure the following sentences are **not** marked as passive:
  - `Je mange.` (active sentence without participle)
  - `J'ai mangé.` (active sentence with regular participle)
  - `J'ai vendu.` (active sentence with irregular participle)
  - `J'étais allé.` (active sentence with passive auxiliary and non-passive participle from adjective/verb exception list)
  - `Elle est la sœur aînée.` (active sentence with passive auxiliary and non-passive é-word from adjective/verb exception list plus suffix)
  - `J’étais au café.` (active sentence with passive auxiliary and non-passive é-word from noun exception list)
  - `Ce sont deux cafés sympas.`  (active sentence with passive auxiliary and non-passive é-word from noun exception list plus suffix)
  - `Est-ce que la création de cet outil contribuera à améliorer l’intégrité scientifique ?` (active sentence with with passive auxiliary and non-passive é-word from noun exception list plus contracted article)
  - `Elle est triste malgré tout.`   (active sentence with passive auxiliary and non-passive é-word from "other" exception list)
  - `Ils se sont lavés.` (active sentence with être-auxiliary that's preceded by a reflexive pronoun)
  - `Cela est le film le plus vu.` (active sentence with être-auxiliary that's followed by an article)
  - `Elle est en vue.` (active sentence with a participle that's marked as non-passive by a direct precedence exception)
  - `Il n’est pas nécessaire d’avoir vu le premier film.` (active sentence with a participle that's marked as non-passive by a precedence exception)
  - `Il s'est lavé.` (active sentence with être-auxiliary that's fused with a reflexive pronoun).

@agnieszkaszuba is going to test this assessment more extensively with a wider range of texts before it can be merged.

Fixes #1286

### CODE REVIEW

File: spec/researches/getPassiveVoiceFrenchSpec.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
+var passiveVoice = require( "../../js/researches/getPassiveVoice.js" );
+var Paper = require( "../../js/values/Paper.js" );
+
+describe( "detecting passive voice in sentences", function() {
+
+	it( "returns active voice (présent)", function () {
+		var paper = new Paper("Je mange une pomme." , { locale: "fr_FR" } );
```
Comment (IreneStr) [inline spec/researches/getPassiveVoiceFrenchSpec.js]: Please check spacing:
- Add spaces within parentheses
- Remove spaces after the `function` keword and before commas

File: spec/researches/getPassiveVoiceFrenchSpec.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+	});
+
+	it( "returns passive voice (présent) with an irregular participle", function () {
+		// Passive: est lu.
+		var paper = new Paper("Un libre est lu par moi." , { locale: "fr_FR" } );
+		expect( passiveVoice( paper ).passives.length).toBe( 1 );
+	});
+
+	it( "returns passive voice (présent) with an irregular feminine participle", function () {
+		// Passive: est lu.
```
Comment (IreneStr) [inline spec/researches/getPassiveVoiceFrenchSpec.js]: est vendue

File: spec/researches/getPassiveVoiceFrenchSpec.js [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+		expect( passiveVoice( paper ).passives.length).toBe( 1 );
+	});
+
+	it( "returns passive voice (présent) with an irregular participle", function () {
+		// Passive: est lu.
+		var paper = new Paper("Un libre est lu par moi." , { locale: "fr_FR" } );
+		expect( passiveVoice( paper ).passives.length).toBe( 1 );
+	});
+
+	it( "returns passive voice (présent) with an irregular feminine participle", function () {
```
Comment (IreneStr) [inline spec/researches/getPassiveVoiceFrenchSpec.js]: What about feminine and masculine plurals?

Comment (IreneStr) [reply spec/researches/getPassiveVoiceFrenchSpec.js]: Also for other tenses. + feminine is not covered for all tenses

File: spec/researches/getPassiveVoiceFrenchSpec.js [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+		var paper = new Paper("J'avais mangé une pomme." , { locale: "fr_FR" } );
+		expect( passiveVoice( paper ).passives.length ).toBe( 0 );
+	});
+
+	it( "returns passive voice (plus-que-parfait)", function () {
+		// Passive: avait été mangée.
+		var paper = new Paper("Une pomme avait été mangée par moi." , { locale: "fr_FR" } );
+		expect( passiveVoice( paper ).passives.length).toBe( 1 );
+	});
+} );
```
Comment (IreneStr) [inline spec/researches/getPassiveVoiceFrenchSpec.js]: Missing newline at end of file

File: spec/researches/passivevoice/getParticiplesFrenchSpec.js [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+		var auxiliaries = mockSentence.getAuxiliaries();
+		var foundParticiples = getParticiples( sentencePartText, auxiliaries, "fr" );
+		expect( foundParticiples[ 0 ].getParticiple() ).toEqual( "informé" );
+		expect( foundParticiples[ 0 ].getType() ).toEqual( "regular" );
+		expect( foundParticiples[ 0 ].getSentencePart() ).toEqual( "était-il informé de cela ?" );
+		expect( foundParticiples[ 0 ].getAuxiliaries() ).toEqual( [ "était-il" ] );
+		expect( foundParticiples[ 0 ].determinesSentencePartIsPassive() ).toEqual( true );
+	});
+
+	it("returns matched regular participles and filtered exceptions; exceptions don't set the sentence part passive.", function(){
```
Comment (IreneStr) [inline spec/researches/passivevoice/getParticiplesFrenchSpec.js]: Missing word in final sentence?

File: spec/researches/passivevoice/getParticiplesFrenchSpec.js [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+	});
+
+	it("returns an empty array when there is no participle", function(){
+		var mockSentence = new sentencePart( "Je voulais vous demander pardon.", [], "fr" );
+		var sentencePartText = mockSentence.getSentencePartText();
+		var auxiliaries = mockSentence.getAuxiliaries();
+		var foundParticiples = getParticiples( sentencePartText, auxiliaries, "fr" );
+		expect( foundParticiples ).toEqual( [] );
+	});
+});
```
Comment (IreneStr) [inline spec/researches/passivevoice/getParticiplesFrenchSpec.js]: Missing newline at end of file

File: spec/researches/passivevoice/getSentencePartsEnglishSpec.js [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+		expect( getSentenceParts( sentence, "en" ).length ).toBe( 1 );
+	} );
+	it ( "splits sentences on stop characters when followed by a punctuation mark", function() {
+		var sentence = "\"This is it\", he said.";
+		expect( getSentenceParts( sentence, "en" )[ 0 ].getSentencePartText() ).toBe( "is it\"" );
+    expect( getSentenceParts( sentence, "en" ).length ).toBe( 1 );
+  } );
+	it ( "correctly splits English sentences when matching punctuation after words ending in ing", function() {
+		var sentence = "(is having)";
+		expect( getSentenceParts( sentence, "en" )[ 0 ].getSentencePartText() ).toBe( "having)" );
```
Comment (IreneStr) [inline spec/researches/passivevoice/getSentencePartsEnglishSpec.js]: space after 'having'

File: spec/researches/passivevoice/getSentencePartsFrenchSpec.js [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+	} );
+
+	it ( "returns all sentence parts from the auxiliary to the stopword", function() {
+		var sentence = "Ils étaient très heureux lorsque je les y conduisais.";
+		expect( getSentenceParts( sentence, "fr" )[ 0 ].getSentencePartText() ).toBe( "étaient très heureux" );
+		expect( getSentenceParts( sentence, "fr" ).length ).toBe( 1 );
+	} );
+
+	it ( "doesn't split on sentence breakers within words", function() {
+		var sentence = "Désormais tu es responsable de tes actes.";
```
Comment (IreneStr) [inline spec/researches/passivevoice/getSentencePartsFrenchSpec.js]: Please add comment with the sentence breaker

File: spec/researches/passivevoice/getSentencePartsFrenchSpec.js [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+		expect( getSentenceParts( sentence, "fr" ).length ).toBe( 1 );
+	} );
+
+	it ( "splits sentences on stop characters", function() {
+		var sentence = "C'est en particulier et principalement une question d'argent, a résumé le Premier ministre néerlandais Mark Rutte.";
+		expect( getSentenceParts( sentence, "fr" )[ 0 ].getSentencePartText() ).toBe( "C'est en particulier et principalement une question d'argent" );
+		expect( getSentenceParts( sentence, "fr" ).length ).toBe( 1 );
+	} );
+
+	it ( "doesn't split sentences on stop characters that are not preceded by a word and also not followed by a space/punctuation mark", function() {
```
Comment (IreneStr) [inline spec/researches/passivevoice/getSentencePartsFrenchSpec.js]: The above test has a stop character preceded by a word AND followed by a space/punctuation mark, and this one has neither.
Can you please add two test to cover the situations where one of those criteria is met?

File: spec/researches/passivevoice/getSentencePartsFrenchSpec.js [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+		var sentence = "La branche était déficitaire de 1,5 milliard.";
+		expect( getSentenceParts( sentence, "fr" )[ 0 ].getSentencePartText() ).toBe( "était déficitaire de 1,5 milliard." );
+		expect( getSentenceParts( sentence, "fr" ).length ).toBe( 1 );
+	} );
+
+	it ( "doesn't split sentences when an auxiliary is preceded by a reflexive pronoun", function() {
```
Comment (IreneStr) [inline spec/researches/passivevoice/getSentencePartsFrenchSpec.js]: doesn't return a sentence part when...

File: spec/researches/passivevoice/getSentencePartsFrenchSpec.js [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+		expect( getSentenceParts( sentence, "fr" ).length ).toBe( 0 );
+	} );
+
+	it ( "filters the sentence part when it is included because of a non-auxiliary sentence breaker but contains an auxiliary preceded by a reflexive pronoun", function() {
+		var sentence = "Nous sommes arrivés, nous nous sommes lavés, et puis nous nous sommes couchés.";
+		expect( getSentenceParts( sentence, "fr" )[ 0 ].getSentencePartText() ).toBe( "sommes arrivés" );
+		expect( getSentenceParts( sentence, "fr" ).length ).toBe( 1 );
+	} );
+
+	it ( "doesn't split sentences when an auxiliary is followed by a word from the followingAuxiliaryExceptionWords list", function() {
```
Comment (IreneStr) [inline spec/researches/passivevoice/getSentencePartsFrenchSpec.js]: Please add comment with word from the followingAuxiliaryException list

File: spec/researches/passivevoice/getSentencePartsFrenchSpec.js [hunk 12]
```diff
@@ -0,0 +1,51 @@
...
+		expect( getSentenceParts( sentence, "fr" )[ 0 ].getSentencePartText() ).toBe( "était déficitaire de 1,5 milliard." );
+		expect( getSentenceParts( sentence, "fr" ).length ).toBe( 1 );
+	} );
+
+	it ( "doesn't split sentences when an auxiliary is preceded by a reflexive pronoun", function() {
+		var sentence = "Ils se sont lavés.";
+		expect( getSentenceParts( sentence, "fr" ).length ).toBe( 0 );
+	} );
+
+	it ( "filters the sentence part when it is included because of a non-auxiliary sentence breaker but contains an auxiliary preceded by a reflexive pronoun", function() {
```
Comment (IreneStr) [inline spec/researches/passivevoice/getSentencePartsFrenchSpec.js]: I don't think I understand what you mean by the first part of this sentence 😬

File: spec/researches/passivevoice/matchParticiplesSpecFrench.js [hunk 13]
```diff
@@ -0,0 +1,49 @@
...
+
+	it( "does not match a word that is not an irregular participle", function() {
+		expect( participleRegexes.irregularParticiples( "pomme", "fr" ) ).toEqual( [] );
+	} );
+
+	it( "matches an irregular participle with an irregular conjugation pattern", function() {
+		expect( participleRegexes.irregularParticiples( "mû", "fr" ) ).toEqual( [ "mû" ] );
+	} );
+
+	it( "matches an irregular participle ending in -s", function() {
```
Comment (IreneStr) [inline spec/researches/passivevoice/matchParticiplesSpecFrench.js]: 'promise' doesn't end in -s 🙃

File: src/stringProcessing/directPrecedenceException.js [hunk 14]
```diff
@@ -0,0 +1,30 @@
+var getWordIndices = require( "../researches/passivevoice/getIndicesWithRegex.js" );
```
Comment (IreneStr) [inline src/stringProcessing/directPrecedenceException.js]: I'm not sure whether this is the right location for this file (and related files). Why aren't they in the passive voice folder/language specific folder?

File: src/researches/english/EnglishParticiple.js [hunk 15]
```diff
@@ -82,25 +22,11 @@ var precedesIndex = function( precedingWords, participleIndex ) {
  */
 var EnglishParticiple = function( participle, sentencePart, attributes ) {
 	Participle.call( this, participle, sentencePart, attributes );
-	this.checkException();
+	checkException.call( this );
```
Comment (IreneStr) [inline src/researches/english/EnglishParticiple.js]: Why this change?

File: src/researches/passivevoice/getSentenceParts.js [hunk 16]
```diff
@@ -0,0 +1,253 @@
...
+ * @param {string} language The language for which to match the auxiliaries.
+ * @returns {Array} All formatted matches from the sentence part.
+ */
+var getAuxiliaryMatches = function( sentencePart, language ) {
+	var auxiliaryRegex = languageVariables[ language ].auxiliaryRegex;
+	var auxiliaryMatches = sentencePart.match( auxiliaryRegex ) || [];
+
+	switch( language ) {
+		case "fr":
+			var auxiliaryMatchesOutput = [];
```
Comment (IreneStr) [inline src/researches/passivevoice/getSentenceParts.js]: I'm not sure about this name

File: src/researches/passivevoice/getSentenceParts.js [hunk 17]
```diff
@@ -0,0 +1,253 @@
...
+			var auxiliaryMatchIndices = getIndicesOfList( auxiliaryMatches, sentencePart );
+			auxiliaryMatchIndices = auxiliaryPrecedenceException( sentencePart, auxiliaryMatchIndices );
+			auxiliaryMatchIndices = followingAuxiliaryException( sentencePart, auxiliaryMatchIndices );
+
+			forEach( auxiliaryMatchIndices, function( auxiliaryMatchIndex ) {
```
Comment (IreneStr) [inline src/researches/passivevoice/getSentenceParts.js]: Please add inline comment to clarify

File: src/researches/passivevoice/getSentenceParts.js [hunk 18]
```diff
@@ -0,0 +1,253 @@
...
+				auxiliaryMatchesOutput.push( auxiliaryMatchIndex.match );
+			} );
+
+			return map( auxiliaryMatchesOutput, function( auxiliaryMatch ) {
```
Comment (IreneStr) [inline src/researches/passivevoice/getSentenceParts.js]: Please add inline comment to clarify

File: src/researches/passivevoice/matchParticiples.js [hunk 19]
```diff
@@ -0,0 +1,92 @@
...
+	},
+	fr: {
+		regularParticiplesRegex: /\S+(é|ée|és|ées)($|[ \n\r\t\.,'\(\)\"\+\-;!?:\/»«‹›<>])/ig,
+	},
+};
+
+/**
+ * Returns words that have been determined to be a regular participle.
+ *
+ * @param {string} word The word to check
```
Comment (IreneStr) [inline src/researches/passivevoice/matchParticiples.js]: period

File: src/researches/passivevoice/matchParticiples.js [hunk 20]
```diff
@@ -0,0 +1,92 @@
...
+ * @param {string} suffixes The suffixes to match the word with.
+ * @param {Array} matches The array into which to push the matches.
+ * @returns {Array} A list with matched irregular participles.
+ */
+var matchFrenchParticipleWithSuffix = function( word, irregulars, suffixes, matches ) {
+	forEach( irregulars, function( irregular ) {
+		var irregularParticiplesRegex = new RegExp( "^" + irregular + suffixes + "?$", "ig" );
+		matches.push( word.match( irregularParticiplesRegex ) || [] );
+	} );
+	matches = [].concat.apply( [], matches );
```
Comment (IreneStr) [inline src/researches/passivevoice/matchParticiples.js]: Simplify? See above

File: src/researches/passivevoice/matchParticiples.js [hunk 21]
```diff
@@ -0,0 +1,92 @@
...
+/**
+ * Returns an array of matches of irregular participles with suffixes.
+ *
+ * @param {string} word The word to match on.
+ * @param {Array} irregulars The list of irregulars to match.
+ * @param {string} suffixes The suffixes to match the word with.
+ * @param {Array} matches The array into which to push the matches.
+ * @returns {Array} A list with matched irregular participles.
+ */
+var matchFrenchParticipleWithSuffix = function( word, irregulars, suffixes, matches ) {
```
Comment (IreneStr) [inline src/researches/passivevoice/matchParticiples.js]: Why do you pass matches?

File: src/researches/passivevoice/matchParticiples.js [hunk 22]
```diff
@@ -0,0 +1,92 @@
...
+ */
+var irregularParticiples = function( word, language ) {
+	var matches = [];
+	switch ( language ) {
+		case "fr":
+			// Match different classes of participles with suffixes.
+			matches = matchFrenchParticipleWithSuffix( word, irregularsRegularFrench, "(e|s|es)", matches );
+			matches = matchFrenchParticipleWithSuffix( word, irregularsEndingInSFrench, "(e|es)", matches );
+			// Match irregular participles that don't require adding a suffix.
+			find( irregularsIrregularFrench, function( currentWord ) {
```
Comment (IreneStr) [inline src/researches/passivevoice/matchParticiples.js]: Please rename 'currentWord' to 'irregularParticiple'

File: src/researches/passivevoice/matchParticiples.js [hunk 23]
```diff
@@ -0,0 +1,92 @@
...
+				if( currentWord === word ) {
+					matches.push( currentWord );
+				}
+			} );
+			break;
+		case "en":
+		default:
+			find( irregularsEnglish, function( currentWord ) {
```
Comment (IreneStr) [inline src/researches/passivevoice/matchParticiples.js]: Please rename 'currentWord' to 'irregularParticiple'

File: src/stringProcessing/followsIndex.js [hunk 24]
```diff
@@ -0,0 +1,26 @@
+var isEmpty = require( "lodash/isEmpty" );
+var forEach = require( "lodash/forEach" );
+var includes = require( "lodash/includes" );
+
+/**
+ *
```
Comment (IreneStr) [inline src/stringProcessing/followsIndex.js]: Missing description

File: src/stringProcessing/precedesIndex.js [hunk 25]
```diff
@@ -0,0 +1,35 @@
...
+
+	var matches = [];
+	forEach( precedingWordsIndices, function( precedingWordsIndex ) {
+		// + 1 because the beginning word boundary is not included in the passive participle match
+		if ( precedingWordsIndex + 1 < participleIndex ) {
+			matches.push( precedingWordsIndex );
+		}
+	} );
+
+	if ( matches.length ) {
```
Comment (IreneStr) [inline src/stringProcessing/precedesIndex.js]: This might be personal, but I think `if ( matches.length > 0) {` is more readable.

File: spec/researches/french/FrenchParticipleSpec.js [hunk 26]
```diff
@@ -0,0 +1,119 @@
...
+		expect( mockParticiple.getParticiple() ).toBe( "malgré" );
+		expect( mockParticiple.isOnAdjectivesVerbsExceptionList() ).toBe( false );
+		expect( mockParticiple.isOnNounsExceptionList() ).toBe( false );
+		expect( mockParticiple.isOnOthersExceptionList() ).toBe( true );
+		expect( mockParticiple.directPrecedenceException( mockParticiple._sentencePart, 12 ) ).toBe( false );
+		expect( mockParticiple.precedenceException( mockParticiple._sentencePart, 12 ) ).toBe( false );
+		expect( mockParticiple.determinesSentencePartIsPassive() ).toBe( false );
+	});
+
+	it( "checks the properties of the French participle object with an adjective or verb exception ending in é plus suffix", function() {
```
[the remark this answers is not in the document]
Comment (manuelaugustin) [reply spec/researches/french/FrenchParticipleSpec.js]: Removed the test because it was a duplicate of another test above.

File: src/researches/english/EnglishParticiple.js (hunk 15 above)
Comment (manuelaugustin) [reply src/researches/english/EnglishParticiple.js]: Before the change, the function `checkException` had been defined as a method of the `EnglishParticiple` object prototype. This function has now been moved to an external file because it’s also used for the French participle. In order to make the `this` in the `checkException` function definition refer to `EnglishParticiple`, `call( this )` was used to pass the correct reference for `this` (the current object) when calling the function.

File: src/researches/french/FrenchParticiple.js [hunk 27]
```diff
@@ -0,0 +1,152 @@
+var Participle = require( "../../values/Participle.js" );
```
[the remark this answers is not in the document]
Comment (manuelaugustin) [reply src/researches/french/FrenchParticiple.js]: It's also used for the other languages.

File: src/stringProcessing/directPrecedenceException.js (hunk 14 above)
Comment (manuelaugustin) [reply src/stringProcessing/directPrecedenceException.js]: You mean `directPrecedenceException.js`? I put it in this folder because it's used for both English and French.

File: src/researches/french/FrenchParticiple.js [hunk 28]
```diff
@@ -0,0 +1,152 @@
...
+ * If a word is on the list, it isn't a participle.
+ * Irregular participles do not end in -é and therefore can't be on the list.
+ *
+ * @returns {boolean} Returns true if it is in the exception list of adjectives and verbs, otherwise returns false.
+ */
+FrenchParticiple.prototype.isOnAdjectivesVerbsExceptionList = function() {
+	checkIrregular.call( this );
+
+	var participle = this.getParticiple();
+	var matches = [];
```
[the remark this answers is not in the document]
Comment (manuelaugustin) [reply src/researches/french/FrenchParticiple.js]: You’re right, there can only be one match. I’ve renamed the variable accordingly.

**Comment by IreneStr** [comment]
CR2 done 👍

File: spec/researches/getPassiveVoiceFrenchSpec.js [hunk 29]
```diff
@@ -73,4 +67,148 @@ describe( "detecting passive voice in sentences", function() {
...
+	});
+
+	it( "returns active voice with a participle that's marked as non-passive by a precedence exception (directly preceding).", function() {
+		// Precedence exception word: avoir (in between "n'est" and "vu").
+		var paper = new Paper( "Il n’est pas nécessaire d’avoir vu le premier film.", { locale: "fr_FR" } );
+		expect( passiveVoice( paper ).passives.length).toBe( 0 );
+	});
+
+	it( "returns active voice with an être-auxiliary that's marked as non-passive by an elision precedence exception.", function() {
+		// Elision precedence exception word: 's.
```
Comment (IreneStr) [inline spec/researches/getPassiveVoiceFrenchSpec.js]: s'

File: spec/researches/french/passiveVoice/FrenchParticipleSpec.js [hunk 30]
```diff
@@ -86,7 +86,14 @@ describe( "A test for checking the French participle", function() {
 		expect( mockParticiple.determinesSentencePartIsPassive() ).toBe( false );
 	});
 
-	it( "checks the properties of the French participle object with a precedence exception", function() {
+	it( "checks the properties of the French participle object with a precedence exception (indirectly preceding)", function() {
```
Comment (IreneStr) [inline spec/researches/french/passiveVoice/FrenchParticipleSpec.js]: Please comment the exception

File: spec/researches/french/passiveVoice/FrenchParticipleSpec.js [hunk 31]
```diff
@@ -86,7 +86,14 @@ describe( "A test for checking the French participle", function() {
...
+		let mockParticiple = new FrenchParticiple( "vu", "n'est pas possible de l'avoir déjà vu.", { auxiliaries: [ "n'est" ], type: "irregular" } );
+		expect( mockParticiple.directPrecedenceException( mockParticiple._sentencePart, 35 ) ).toBe( false );
+		expect( mockParticiple.precedenceException( mockParticiple._sentencePart, 35 ) ).toBe( true );
+		expect( mockParticiple.determinesSentencePartIsPassive() ).toBe( false );
+	});
+
+	it( "checks the properties of the French participle object with a precedence exception (directly preceding)", function() {
```
Comment (IreneStr) [inline spec/researches/french/passiveVoice/FrenchParticipleSpec.js]: Please comment the exception

**Comment by IreneStr** [comment]
Acceptance done 👍
